blaze/compute/core.py

```diff
@@ -446,12 +446,13 @@ def compute(expr, d, **kwargs):
     optimize_ = kwargs.get('optimize', optimize)
     pre_compute_ = kwargs.get('pre_compute', pre_compute)
     post_compute_ = kwargs.get('post_compute', post_compute)
-
     expr2, d2 = swap_resources_into_scope(expr, d)
     if pre_compute_:
-        d3 = dict([(e, pre_compute_(expr2, dat, **kwargs))
```

Comment:
It looks like most cases of `pre_compute` don't care about the expression being passed in so changing the input expr to be the leaf would probably be safe.  At this point we would probably want to just kill the expression as an input though.